post/clang_tidy_review/clang_tidy_review/__init__.py

```diff
@@ -60,6 +61,53 @@ class PRReview(TypedDict):
     comments: list[ReviewComment]
 
 
+# These types are from include/clang/Tooling/DiagnosticsYaml.h
+# and include/clang/Tooling/ReplacementsYaml.h
+class ClangFileByteRange(TypedDict):
```

Comment:
Why didn't you use `@dataclass`?

Reply:
We get these from Clang by parsing YAML. And PyYAML parses to dictionaries. If we wanted to use dataclasses, we'd need to convert each record (and we'd need to do that recursively). `TypedDict` has no runtime overhead and just informs the type system about the dictionary's elements.